code/modules/reagents/reagent_containers/mortar.dm

```diff
@@ -33,41 +36,81 @@ to accommodate additional materials.
-			if((do_after(user, 25, target = src)) && grinded)
-				user.adjustStaminaLoss(40)
-				if(grinded.juice_results) //prioritize juicing
-					grinded.on_juice()
-					reagents.add_reagent_list(grinded.juice_results)
-					to_chat(user, "<span class='notice'>You juice [grinded] into a fine liquid.</span>")
-					QDEL_NULL(grinded)
-					return
-				grinded.on_grind()
-				reagents.add_reagent_list(grinded.grind_results)
-				if(grinded.reagents) //food and pills
-					grinded.reagents.trans_to(src, grinded.reagents.total_volume, transfered_by = user)
-				to_chat(user, "<span class='notice'>You break [grinded] into powder.</span>")
-				QDEL_NULL(grinded)
-				return
+	if(istype(attacking_item, /obj/item/pestle))
+		if(!grinded)
+			balloon_alert(user, "nothing to grind")
 			return
-		else
-			to_chat(user, "<span class='warning'>There is nothing to grind!</span>")
+
+		if(user.getStaminaLoss() > MORTAR_STAMINA_MINIMUM)
+			balloon_alert(user, "too tired")
 			return
-	if(grinded)
-		to_chat(user, "<span class='warning'>There is something inside already!</span>")
+
+		var/list/choose_options = list(
+			"Grind" = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_grind"),
+			"Juice" = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_juice")
+		)
+		var/picked_option = show_radial_menu(user, src, choose_options, radius = 38, require_near = TRUE)
+
+		if(!grinded || !in_range(src, user) || !user.is_holding(attacking_item) && !picked_option)
+			return
+
+		balloon_alert(user, "grinding...")
+		if(!do_after(user, grind_speed, target = src))
+			balloon_alert(user, "stopped grinding")
+			return
+
+		user.adjustStaminaLoss(MORTAR_STAMINA_USE)
+		switch(picked_option)
+			if("Juice")
+				if(grinded.juice_results)
+					juice_target_item(grinded, user)
+				else
+					grind_target_item(grinded, user)
+
+			if("Grind")
+				if(grinded.grind_results)
+					grind_target_item(grinded, user)
+				else
+					juice_target_item(grinded, user)
 		return
-	if(I.juice_results || I.grind_results)
-		I.forceMove(src)
-		grinded = I
+
+	if(!attacking_item.juice_results && !attacking_item.grind_results)
+		balloon_alert(user, "can't grind this")
+		return ..()
+
+	if(grinded)
+		balloon_alert(user, "already full")
 		return
-	to_chat(user, "<span class='warning'>You can't grind this!</span>")
+
+	attacking_item.forceMove(src)
+	grinded = attacking_item
+
+///Juices the passed target item, and transfers any contained chems to the mortar as well
+/obj/item/reagent_containers/glass/mortar/proc/juice_target_item(obj/item/to_be_juiced, mob/living/carbon/human/user)
+	to_be_juiced.on_juice()
+	reagents.add_reagent_list(to_be_juiced.juice_results)
+
+	if(to_be_juiced.reagents) //If juiced item has reagents within, transfer them to the mortar
+		to_be_juiced.reagents.trans_to(src, to_be_juiced.reagents.total_volume, transfered_by = user)
+
+	to_chat(user, span_notice("You juice [to_be_juiced] into a fine liquid."))
+	QDEL_NULL(to_be_juiced)
```

Comment:
This doesn't actually set the object's "grinded" variable to null, just the proc-local variable "to_be_juiced". This -- storing references to objects after they are deleted -- is what causes harddeletes.
To fix this, just add "grinded = null" after juicing / grinding; this change should also be made to the large mortar.

Reply:
This is a copied behavior from large mortars i believe. Will fix that for that as well